openvino_xai/methods/black_box/aise.py

```diff
@@ -0,0 +1,221 @@
+# Copyright (C) 2024 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+import collections
+from typing import Callable, List
+
+import math
+from scipy.optimize import direct, Bounds
+import numpy as np
+import openvino.runtime as ov
+from openvino.runtime.utils.data_helpers.wrappers import OVDict
+
+from openvino_xai.common.utils import IdentityPreprocessFN, infer_size_from_image, scaling, sigmoid
+from openvino_xai.methods.black_box.base import BlackBoxXAIMethod, Preset
+
+
+class GaussianPerturbationMask:
+    def __init__(self, input_size):
+        h = np.linspace(0, 1, input_size[0])
+        w = np.linspace(0, 1, input_size[1])
+        self.h, self.w = np.meshgrid(w, h)
+
+    def _gaussian_2d(self, gauss_params):
+        mh, mw, sigma = gauss_params
+        A = 1 / (2 * math.pi * sigma * sigma)
+        B = (self.h - mh) ** 2 / (2 * sigma ** 2)
+        C = (self.w - mw) ** 2 / (2 * sigma ** 2)
+        return A * np.exp(-(B + C))
+
+    def generate_kernel_mask(self, gauss_param, scale=1.0):
+        kernel = self._gaussian_2d(gauss_param)
+        return (kernel / kernel.max()) * scale
+
+
+class AISE(BlackBoxXAIMethod):
+    """AISE explains classification models in black-box mode using 
+    AISE: Adaptive Input Sampling for Explanation of Black-box Models.
```

Comment:
The link to the paper could be useful here

Reply:
It is not published yet, will add later